### PR TITLE
Support generic postgres custom metrics

### PR DESCRIPTION
**Pull request by sethcleveland**

### CODE REVIEW

**Comment by sethcleveland** [comment]
Dropping in puppet support for the following -- https://help.datadoghq.com/hc/en-us/articles/208385813-Postgres-custom-metric-collection-explained

File: manifests/integrations/postgres.pp [hunk 1]
```diff
@@ -19,6 +19,12 @@
 #       Track per relation/table metrics. Array of strings.
 #       Warning: this can collect lots of metrics per relation
 #       (10 + 10 per index)
+#   $tags
+#       Optional array of tags
+#   $custom_metrics
+#       An array of custom metrics with the following hash keys - query, metrics,
+#       relation, descriptors. Refer to this guide for details on those fields:
+#       https://help.datadoghq.com/hc/en-us/articles/208385813-Postgres-custom-metric-collection-explained
```
Comment (truthbk) [inline manifests/integrations/postgres.pp]: Could you please add an example with custom metrics below? Should help a bit with usability.

**Comment by truthbk** [comment]
Thanks a lot @sethcleveland, tackles an open issue which is always greatly appreciated! If you could add some spec tests that'd be fabulous.

**Comment by sethcleveland** [comment]
Sure thing. This is an open issue for us as well. I'll add documentation and spec tests.

**Comment by sethcleveland** [comment]
I did a brief scan on the failures and everything appears to fail this --
`bundle exec rake test
---> syntax:manifests
---> syntax:templates
---> syntax:hiera:yaml
manifests/integrations/mysql.pp:41:parameter_order:WARNING:optional parameter listed before required parameter
manifests/integrations/pgbouncer.pp:31:parameter_order:WARNING:optional parameter listed before required parameter
manifests/integrations/postgres.pp:56:parameter_order:WARNING:optional parameter listed before required parameter`

**Comment by sethcleveland** [comment]
@truthbk: How are these changes?

**Comment by truthbk** [comment]
@sethcleveland this is great, thanks a lot. I'll review in a little bit more depth before merging, but it looks good. I'll keep you posted! I need to understand where those failures are coming from. Maybe if you rebase to the latest `master` we can get rid of them..... :)

**Comment by sethcleveland** [comment]
@truthbk just rebased against master. Hopefully that fixes the problems.

**Comment by sethcleveland** [comment]
@truthbk turns out rebasing against master didn't resolve the failures. I'm assuming the linters were added after the fact. I fixed the failing builds as part of this PR commit -- https://github.com/DataDog/puppet-datadog-agent/pull/224/commits/5cd38a4f1cd306a12eed43aca20f8fe81f12b4c8.

FWIW's I'm not clear why parameter optional/non-optional parameter ordering matters. I've only see puppet modules used with explicit parameter values.

**Comment by rooprob** [comment]
Hi, 

Can we split out and merge https://github.com/DataDog/puppet-datadog-agent/pull/224/commits/5cd38a4f1cd306a12eed43aca20f8fe81f12b4c8 as it's blocking PRs such as https://github.com/DataDog/puppet-datadog-agent/pull/231 and https://github.com/DataDog/puppet-datadog-agent/pull/229.

**Comment by sethcleveland** [comment]
@truthbk do these changes work? Other people can be unblocked by one of the commits. I can cherry pick the fix out into another PR while you spend more time reviewing these changes. Let me know where you'll stand before I make another PR.

**Comment by sethcleveland** [comment]
@rooprob I cherry-picked that commit and made a PR here -- https://github.com/DataDog/puppet-datadog-agent/pull/232.

File: manifests/integrations/postgres.pp [hunk 2]
```diff
@@ -11,14 +13,18 @@
...
 #   $tags
 #       Optional array of tags
 #   $tables
 #       Track per relation/table metrics. Array of strings.
 #       Warning: this can collect lots of metrics per relation
 #       (10 + 10 per index)
+#   $tags
+#       Optional array of tags
+#   $custom_metrics
+#       An array of custom metrics with the following hash keys - query, metrics,
```
Comment (mcasper) [inline manifests/integrations/postgres.pp]: Isn't this a hash of custom metrics?

Comment (sethcleveland) [inline manifests/integrations/postgres.pp]: Yup. I'll make an adjustment.

**Comment by truthbk** [comment]
@sethcleveland thank you so much for this and #232 - merging this in. Again, apologies for the long MTTR.

**Pull request merged**